BankChain/app/src/androidTest/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTest.java

```diff
@@ -144,4 +153,100 @@ public void testExceptionConfirmationNotFuture() {
 
         assertTrue(bank.confirmException(thr) == thr);
     }
+
+    @Test
+    public void testGetUsers() throws ExecutionException, InterruptedException {
+        CompletableFuture<Void> future;
+        Bank bank;
+        Environment env;
+
+        env = new Environment();
+        env.setBank("Bunq");
+
+        env.setBankUrl("REAL_BUNQ_URL");
+        env.setBankApiKey("REAL_BUNQ_API_KEY");
+
+        List<Party> users;
+        bank = new BankFactory(env).create();
+        bank.createSession().get();
+        users = bank.listUsers().get();
+        Party firstUser = users.get(0);
+        assertEquals(firstUser.getId(), 2002);
+        assertEquals(firstUser.getName(), "Overton Onderlinge Waarborgmaatschappij");
+    }
+
+
+    @Test
+    public void testGetAccounts() throws ExecutionException, InterruptedException {
+        CompletableFuture<Void> future;
+        Bank bank;
+        Environment env;
+
+        env = new Environment();
+        env.setBank("Bunq");
+
+        env.setBankUrl("REAL_BUNQ_URL");
+        env.setBankApiKey("REAL_BUNQ_API_KEY");
+
+        bank = new BankFactory(env).create();
+        bank.createSession().get();
+
+        List<Account> accounts = bank.listAccount(new BunqParty("Overton Onderlinge Waarborgmaatschappij", 2002)).get();
+        Account ac = accounts.get(0);
+        assertEquals(ac.getParty().getId(), 2002);
+        assertEquals(ac.getIban(), "NL05BUNQ9900019989" +
+                "");
+        assertEquals(ac.getId(), 2021);
+    }
+
+    @Test
+    public void testListTransactions() throws ExecutionException, InterruptedException {
+        CompletableFuture<Void> future;
+        Bank bank;
+        Environment env;
+
+        env = new Environment();
+        env.setBank("Bunq");
+
+        env.setBankUrl("REAL_BUNQ_URL");
+        env.setBankApiKey("REAL_BUNQ_API_KEY");
+
+        bank = new BankFactory(env).create();
+        bank.createSession().get();
+
+        BunqParty party = new BunqParty("Overton Onderlinge Waarborgmaatschappij", 2002);
+        BunqAccount account = new BunqAccount("NL05BUNQ9900019989", 2021, party);
+
+        List<Transaction> transactions = bank.listTransactions(account).get();
+        Transaction t = transactions.get(transactions.size() - 2);
```

Comment:
The list size constantly changes and can't be independently verified. Only thing that can be done is that the size is at least 3. But if that is not the case, get() will throw an exception and thus also fail the test.